### ISSUE TITLE
Getting error while building from source

### ISSUE THREAD

**Issue by tzpranto** [misc0110/PTEditor#46]
make command giving the following error

> cd module && make
> make[1]: Entering directory '/scratch/mpt5763/Desktop/Experiments/CollidePower/E1_and_E2/PTEditor/module'
> make -C /lib/modules/6.8.0-49-generic/build M=/scratch/mpt5763/Desktop/Experiments/CollidePower/E1_and_E2/PTEditor/module modules
> make[2]: Entering directory '/usr/src/linux-headers-6.8.0-49-generic'
> warning: the compiler differs from the one used to build the kernel
>   The kernel was built by: x86_64-linux-gnu-gcc-13 (Ubuntu 13.2.0-23ubuntu4) 13.2.0
>   You are using:           gcc-13 (Ubuntu 13.2.0-23ubuntu4) 13.2.0
>   CC [M]  /scratch/mpt5763/Desktop/Experiments/CollidePower/E1_and_E2/PTEditor/module/pteditor.o
> /scratch/mpt5763/Desktop/Experiments/CollidePower/E1_and_E2/PTEditor/module/pteditor.c: In function ‘_invalidate_tlb’:
> /scratch/mpt5763/Desktop/Experiments/CollidePower/E1_and_E2/PTEditor/module/pteditor.c:158:7: warning: unused variable ‘pcid’ [-Wunused-variable]
>   158 |   int pcid;
>       |       ^~~~
> In file included from ./arch/x86/include/asm/thread_info.h:53,
>                  from ./include/linux/thread_info.h:60,
>                  from ./include/linux/spinlock.h:60,
>                  from ./include/linux/kref.h:16,
>                  from ./include/linux/mm_types.h:8,
>                  from /scratch/mpt5763/Desktop/Experiments/CollidePower/E1_and_E2/PTEditor/module/pteditor.c:1:
> /scratch/mpt5763/Desktop/Experiments/CollidePower/E1_and_E2/PTEditor/module/pteditor.c: In function ‘pteditor_init’:
> /scratch/mpt5763/Desktop/Experiments/CollidePower/E1_and_E2/PTEditor/module/pteditor.c:709:28: error: ‘X86_FEATURE_INVPCID_SINGLE’ undeclared (first use in this function); did you mean ‘X86_FEATURE_INVPCID’?
>   709 |   if (!cpu_feature_enabled(X86_FEATURE_INVPCID_SINGLE)) {
>       |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~
> ./arch/x86/include/asm/cpufeature.h:144:31: note: in definition of macro ‘cpu_feature_enabled’
>   144 |         (__builtin_constant_p(bit) && DISABLED_MASK_BIT_SET(bit) ? 0 : static_cpu_has(bit))
>       |                               ^~~
> /scratch/mpt5763/Desktop/Experiments/CollidePower/E1_and_E2/PTEditor/module/pteditor.c:709:28: note: each undeclared identifier is reported only once for each function it appears in
>   709 |   if (!cpu_feature_enabled(X86_FEATURE_INVPCID_SINGLE)) {
>       |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~
> ./arch/x86/include/asm/cpufeature.h:144:31: note: in definition of macro ‘cpu_feature_enabled’
>   144 |         (__builtin_constant_p(bit) && DISABLED_MASK_BIT_SET(bit) ? 0 : static_cpu_has(bit))
>       |                               ^~~
> make[4]: *** [scripts/Makefile.build:243: /scratch/mpt5763/Desktop/Experiments/CollidePower/E1_and_E2/PTEditor/module/pteditor.o] Error 1
> make[3]: *** [/usr/src/linux-headers-6.8.0-49-generic/Makefile:1925: /scratch/mpt5763/Desktop/Experiments/CollidePower/E1_and_E2/PTEditor/module] Error 2
> make[2]: *** [Makefile:240: __sub-make] Error 2
> make[2]: Leaving directory '/usr/src/linux-headers-6.8.0-49-generic'
> make[1]: *** [Makefile:5: all] Error 2
> make[1]: Leaving directory '/scratch/mpt5763/Desktop/Experiments/CollidePower/E1_and_E2/PTEditor/module'
> make: *** [Makefile:9: pteditor] Error 2
> 
> >

**Comment by cfied** [comment]
X86_FEATURE_INVPCID_SINGLE has been deprecated with linux version 6.7-rc4

Effectively INVPCID && CR4.PCIDE=1

**Comment by misc0110** [comment]
Yes, but this has been fixed in https://github.com/misc0110/PTEditor/commit/75d3c7f390afe039da9e0775dc2f77b670a719b1 already
